templates/ecs.yaml

```diff
@@ -0,0 +1,458 @@
+    Properties:
+      Certificates:
+        - CertificateArn: !Ref CertificateArn
+      ListenerArn: !Ref AlbHttpsListener
+
+  # ALB Load Balancer
+  EcsAlb:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      Scheme: internet-facing
+      LoadBalancerAttributes:
+        - Key: idle_timeout.timeout_seconds
+          Value: '30'
+      Subnets: !Ref 'PublicSubnetIds'
+      SecurityGroups: [!Ref 'EcsAlbSecurityGroup']
+  # The LoadBalancer listener for HTTP redirects to HTTPS.
+  AlbHttpListener:
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    DependsOn: EcsServiceRole
+    Properties:
+      DefaultActions:
+        - Order: 1
+          Type: 'redirect'
+          RedirectConfig:
+            Protocol: 'HTTPS'
+            Port: '443'
+            Host: '#{host}'
+            Path: '/#{path}'
+            Query: '#{query}'
+            StatusCode: 'HTTP_301'
+      LoadBalancerArn: !Ref 'EcsAlb'
+      Port: 80
+      Protocol: 'HTTP'
+  # The HTTPS loadBalancer listener does the actual forwarding.
+  AlbHttpsListener:
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    DependsOn: EcsServiceRole
+    Properties:
+      DefaultActions:
+        - Order: 1
+          Type: 'forward'
+          TargetGroupArn: !Ref 'EcsTargetGroup'
+      LoadBalancerArn: !Ref 'EcsAlb'
+      Port: 443
+      Protocol: 'HTTPS'
+      SslPolicy: 'ELBSecurityPolicy-2016-08'
+      Certificates:
+        - CertificateArn: !Ref CertificateArn
+  # Associate the domain name with the loadBalancer.
+  iAtlasDNS:
+    Type: AWS::Route53::RecordSetGroup
+    Properties:
+      HostedZoneId: !Ref HostedZone
+      Comment: Zone apex alias targeted to EcsAlb LoadBalancer.
+      RecordSets:
+        - Name: !Sub '${SubDomainName}.${DomainName}'
+          Type: A
+          AliasTarget:
+            HostedZoneId: !GetAtt 'EcsAlb.CanonicalHostedZoneID'
+            DNSName: !GetAtt 'EcsAlb.DNSName'
+  ContainerInstances:
+    Type: AWS::AutoScaling::LaunchConfiguration
+    Properties:
+      ImageId: !Ref 'LatestEcsAmiId'
+      SecurityGroups:
+        - !Ref 'EcsSecurityGroup'
+        - !Ref 'DbAccessSecurityGroup'
+      InstanceType: !Ref 'InstanceType'
+      IamInstanceProfile: !Ref 'EC2InstanceProfile'
+      KeyName: !Ref 'KeyName'
+      UserData:
+        Fn::Base64: !Sub |
+          #!/bin/bash -xe
+          echo ECS_CLUSTER=${EcsCluster} >> /etc/ecs/ecs.config
+          echo ECS_ENABLE_AWSLOGS_EXECUTIONROLE_OVERRIDE=true >> /etc/ecs/ecs.config
+          yum install -y aws-cfn-bootstrap
+          /opt/aws/bin/cfn-signal -e $? --stack ${AWS::StackName} --resource EcsAutoScalingGroup --region ${AWS::Region}
+  EcsService:
+    Type: AWS::ECS::Service
+    DependsOn:
+      - AlbHttpListener
+      - AlbHttpsListener
+    Properties:
+      Cluster: !Ref 'EcsCluster'
+      DesiredCount: 1
+      LoadBalancers:
+        - ContainerName: iatlas-api
+          ContainerPort: 80
```

Comment:
Possible to send alb->container on 443 with presenting a self-signed cert? The account this would run in has that requirement

Reply:
That is currently out of scope for what we were building. We would be happy to build that if we extend the scope.